### ISSUE TITLE
You should have to fight the crew to plunder internal outfits like reactors

### ISSUE THREAD

**Issue by Pointedstick** [endless-sky/endless-sky#3346]
Considering that the boarding panel has a way to fight the crew to gain control of their ship, it's always felt _really weird_ that you would be able to plunder outfits that are _inside_ the ship without having to fight them first.

It makes sense that external outfits like guns, turrets, engines, and sensors would be plunderable without having to fight the crew; you could remove them (more or less) intact without having to enter the ship. But batteries, bunk rooms, coolers, and _especially_ reactors? To get those out, you'd need to wander the halls with cutting torches and power saws, and the crew would undoubtedly  resist.

We already have an `unplunderable` tag that applies to outfits that can't be plundered without capturing the ship; perhaps we could simply apply this tag to all plausible "internal" outfits.

This would provide a really simple fix to the unfairly strong moneymaking approach of looting big ships' reactors in systems with a lot of pirate activity using a ship with a large cargo hold. Depreciation reduced the effectiveness of this exploit, but it's still there, especially for large navy and pirate ships sporting Armageddon Cores and other huge reactors.

**Comment by Pointedstick** [comment]
I previously filed a similar issue: https://github.com/endless-sky/endless-sky/issues/227

It's still a problem. Less of one, with depreciation, but still there.

**Comment by tehhowch** [comment]
So just boarding a disabled ship would show only a partial list of plunderable outfits? Would there be some way to gain access to `internal` outfits through combat? Would that way be any more logical than the current behavior? ("They retreated so we can stop fighting now!" -> there is still crew preventing you from taking internal outfits)

This unfortunately makes capturing even more preferred than it already is, because plundering as a moneymaker makes no sense once you've defeated the opposing crew unless you can't jump the ship to an outfitter.

**Comment by Pointedstick** [comment]
> So just boarding a disabled ship would show only a partial list of plunderable outfits?

If reactors et al. were marked as `unplunderable`, then yes.

> Would there be some way to gain access to internal outfits through combat? Would that way be any more logical than the current behavior? ("They retreated so we can stop fighting now!" -> there is still crew preventing you from taking internal outfits)

That's an option, just an idea. But IMHO it is vastly more logical than right now, when you can capture a ship's main reactor, hyperdrive, batteries, and cooling systems at zero risk to your own crew, without the enemy crew putting up a fight. What are they doing? Just letting you walk around the ship with cutting torches and power saws?

> This unfortunately makes capturing even more preferred than it already is, because plundering as a moneymaker makes no sense once you've defeated the opposing crew unless you can't jump the ship to an outfitter.

That's fine, because capturing is harder than plundering. It requires a large crew and specialized weapons. This demands purposeful investment in same and an ongoing money drain to support them. The idea is to make it harder to strip a disabled ship of its most valuable outfits without first investing in making your own ship and crew complement support such a thing.

**Comment by jafdy** [comment]
It would make it hard to get jump drives beyond the first.

**Comment by Pointedstick** [comment]
As it logically should be, no? :smiley:

**Comment by jafdy** [comment]
No, because it would be a bad thing for the new player game play experience.

**Comment by Amazinite** [comment]
Remember that in the Checkmate route, players must acquire a jump drive by attacking the Pug.

**Comment by flaviojs** [comment]
I think having to fight for some outfits is fair and more realistic, but there MUST be a visual hint that plundering it will trigger a fight.

Picturing it... I imagine plundering an internal outfit would trigger a round of battle, like a press to Attack/Defend. Which implies the ship crew will decide to attack or defend (no crew damage) based on your battle strength. Since you would be carrying equipment around and/or part of the crew would be busy, your battle strength should be a reduced amount of your attack strength, maybe to half or based on the arguments of the keyword that marks the outfit as internal.

**Comment by ghoulavenger** [comment]
I'm okay with some outfits being unplunderable. But really resistance of the crew you speak of is a bunch of people with H2H weapons taking potshots at your ship. Why do I say this? Because you're not going to board their ship to take their outfits, you're going to strip them straight out of the hull. It might have been internal before, but not after you're done with it. Imho if you wanted more realism you'd require a ship to be outfit scanned before it could be plundered, so you knew where to cut.

**Comment by KiLEdEnNis** [comment]
I'm a bit worried that this will just make it harder for plundering players while capturing players won't get hit at all, which brings me to the gameplay over realism concept

**Comment by Pointedstick** [comment]
> No, because it would be a bad thing for the new player game play experience.

New players don't farm jump drives. :slightly_smiling_face:

**Comment by KiLEdEnNis** [comment]
new players would have a way harder time plundering which doesn't seem good to me, while the privateer doesn't care at all
as for experienced players, they'd just buy a Bactrian w/ a load of bunks & capture Raiders instead to get their Jump Drives, ya just make it a little bit harder
so again: I don't think this change is worth it

**Comment by Pointedstick** [comment]
> new players would have a way harder time plundering 

That's the whole point; it's currently too easy a source of quick cash. You can wander into a battle scene, strip the losers of their reactors and be rich in a game week. This suggestion doesn't make plundering that much harder, since there are still plenty of juicy externally-mounted outfits (Heavy Laser turrets, Particle Cannons, etc). It would just prevents you from amassing wealth excessively and unfairly quickly by selling an infinite supply of looted Fission reactors on Pirate Argosies (among other options).

**Comment by ghoulavenger** [comment]
Speaking as someone who has captured very close to the beginning of the game with a star barge, you're right, that capturing and plundering ships early on is ridiculously good money. However with the time investment in being shot down, and waiting for an opportunity to board a ship will take a great deal of RL time, even with fast forward (navy shoots down not disables most targets, and merchants tend to scoop up the rest). What you're suggesting affects more than salvagers though, it directly affects combat fleets, and combat fleets are by no means the OP way to make money.

Those are the most definitely the trading fleets. Owing largely to the fact that they are infinitely extensible and require very little RL time investment. They just don't have much of an early game (money doesn't start pouring in until you've already got a few freighters).

So the question is, does the in game date matter? If it does, then you're right and plundering and capturing are both too powerful. If the in game date doesn't, then you're absolutely worried about the wrong thing.

**Comment by Pointedstick** [comment]
@ghoulavenger I completely agree, and this suggests that pirates aren't sufficiently responsive or dangerous to larger cargo-heavy fleets. That's a separate issue, though. Feel free to file one for it.

**Comment by tehhowch** [comment]
Boarding a pirate at all (plunder or cap) will be more dangerous if they prefer targeting those who have boarded their allies (see my PR).

**Comment by KiLEdEnNis** [comment]
making plundering impossible for some outfits doesn't seem right to me though
imagine if the player has 0 chance to win a boarding fight (say Sparrow vs Falcon), the defending ship wouldn't give any outfits up, while if the player had a Falcon vs a Sparrow, the crew on the Sparrow would be more like "take the reactor just don't kill us"
what this idea does is: the more "dangerous" your crew seems the more inclined the enemy crew is to not resist when plundering it
and what this hard-to-explain idea does it restrict smaller ships to plundering smaller ships, while bigger ships can freely plunder bigger ones

ik it's a bit poorly explained but the concept is "make the ability to plunder depend on if you could capture the ship"

**Comment by Pointedstick** [comment]
> the crew on the Sparrow would be more like "take the reactor just don't kill us"

Taking their reactor would kill them...

What you're suggesting seems really complicated for little benefit, and is already adequately reflected in the fighting mechanic. If you're so overwhelmingly powerful, you just kill the enemy crew so that you can take their stuff.

A Sparrow boarding a Falcon would and should absolutely be able to cut things off the outside of the Falcon's hull. The ship is disabled, after all. There's no fight left in it unless the Sparrow captain (and possible co-pilot) _enter_ the Falcon.

**Comment by KiLEdEnNis** [comment]
then only being able to loot interior things if ya have the "dominance" would fix it
and as far as ik taking their reactor wouldn't kill them, they're just helpless (while lore-wise others could rescue them off of that ship)
apart from that I do agree with what ya said

**Comment by Pointedstick** [comment]
> and as far as ik taking their reactor wouldn't kill them

I have to imagine that life support systems run on energy. :slightly_smiling_face:

**Comment by MessyMix** [comment]
I'm not sure I agree with this change, but entertaining this idea, why not
have the plunder UI separated into 2 sections - one internal and one
external? The internals would be greyed out, and a tooltip can show if you
hover, saying, "you must board the ship before you can plunder this
outfit". Outfits will be made available to plunder based on the cost of the
outfit vs. the number of crew members remaining. Outfits that are greyed
out start to become available to the player as you kill more and more crew,
pushing them further back into the vessel. Items that remain greyed out
give the tooltip: "Continue attacking to plunder this outfit". This system
works because the reactor is probably the most expensive internal outfit on
the ship, so you'd have to kill pretty much all the crew before you can
take it, and the crew is likely to give up less expensive outfits before
sacrificing their lives. If you want to, say, run away with their jump
drive or shielding matrix, a reasonable crew would give it up before too
many deaths - and this boarding mechanic models this behavior well, since
these outfits are a lot less expensive than the reactor. Just my two cents.

**Comment by Pointedstick** [comment]
I love love love that idea!

**Comment by ghoulavenger** [comment]
@MessyMix at that point why not just capture the ship completely? You're already potentially losing crew as a boarding action, and if there are thresholds to what you can and can't take, you're just filling up your hold needlessly. The only reason not to full capture at this point is because you don't have the crew to maintain both ships.

I still maintain if you're really worried about the internals of a ship you should worry about getting an outfit scanner instead. You're cutting stuff off the ship, if you don't know where to cut it could be more problematic -- like damaging the outfits instead or destroying the ship prematurely. But boarding the ship when your intent is to strip it and not capture it? Doesn't make sense to me.

**Comment by MessyMix** [comment]
Well, it doesn't matter if you know where to cut when you need to get INTO the ship to cut out the reactor, right? And you can't really get into the ship and take the reactor fine and dandy when there are people on the ship who will defend the reactor to their deaths, right?

Gameplay wise, it means that it's a lot less lucrative to plunder reactor than to cap entirely -- but this also makes sense. By the time you have a ship's reactor, you pretty much control the ship. Potential reasons why a player would NOT capture the ship would be if you plundered vital components (i.e. hyperdrive) without an outfitter and then go for the outfitter as well.

Again, I don't know if I'm for or against this change, but if it were to be implemented, I would do it as I described above.

**Comment by flaviojs** [comment]
Exemplifying my earlier suggestion (before and after pressing Take):
![plunder_fight](https://user-images.githubusercontent.com/1009600/33570246-21eaba02-d924-11e7-9e72-6d39806ed1f3.png)
![plunder_fight_result](https://user-images.githubusercontent.com/1009600/33570248-24062a38-d924-11e7-8b30-653ab33bb6f5.png)

By plundering you decided to "attack", but your attack strength is reduced (by half?).
They either decide to attack (fight) or defend (do nothing) during plunder.
The number of rounds is based on their crew size, similar to what happens when we press Attack.
You take 1 outfit is plundered if they fight or the whole stack if they do nothing.

This means you need to "dominate" to plunder safely, but you can still plunder 1 if you have enough crew to last 1 round. Also, your ship can easily become vulnerable to pirates from lack of crew.

The capture/survival odds section would need a rework so there is enough space for plunder odds.

**Comment by ghoulavenger** [comment]
@MessyMix what does the reactor being internal have to do with cutting it out of the hull directly? You just cut through the exterior hull. Sure, that might destroy the ship, but thats perfectly fine, you're basically killing it by taking the reactor anyway.

**Comment by PJTheGuy** [comment]
I think this would be a good idea. It always kind of seems strange when you can steal the reactor while the crew is still there. Either they mount the reactor on the outside of the ship, or the other ship's crew is completely incompetent. I seriously doubt the first one, that's a big fire hazard.

**Comment by Jugosloven1612** [comment]
I agree with ghoulavenger. When you take outfits you are cutting them out with your weapons, not boarding the ship. I like the ide that the ship could be potentialy destroyed with, for example, 20% chance when taking the reactor, if you don't know where to cut. It would also add more use to the outfit scanner.

**Comment by Tadrix** [comment]
Shouldn't the deciding factor for these things be size of the ship and not type of the particular outfit?

Some spacecraft have all compartments accessible from outside. Think Space Shuttle: "The fuel cells are located under the payload bay area in the forward portion of the orbiter's midfuselage." In space it would take an EVA to get to them. This is especially true for smaller vessels (first you get out of the car and then inspect what's under the hood), such as drones, fighters, interceptors, some transports and some light freighters, generally anything with required crew 2 or less. If ship's disabled, everything can be taken.

Some ships have innermost compartments inaccessible from outside (meaning, other compartments are in the way), and this should be true for all capital-size ships. Innermost compartments would usually house most valuable systems, and the main power plant is among them (not necessarily batteries, though).

So, either make unplunderable only outfits above certain size (implying that ship which carries it is built around it), or divide ships into two categories, and for this larger category outfits of certain types would automatically be unplunderable (mainly `category "Power"`).

Cooling systems should be plunderable, their main function is to radiate heat away; radiators usually are embedded in hull.

**Comment by VonZorch** [comment]
All of you keep talking about things being inside the hull.  By the time a ship is being boarded or looted the hull has mostly been shot away.  This leaves any major component easily accessible to your work parties and any approaches to them are covered by anti-ship weapons.

**Comment by Amazinite** [comment]
The hull has been weakened and damaged, but it hasn’t been “shot away” in that the ship is full of holes leading into vital areas of the ship, as otherwise you would expect some of the crew to be dead from being in the wrong room at the wrong time when it depressurizes, which isn’t the case.

**Comment by Zitchas** [comment]
I like the idea of needing an Outfit Scanner to be able to loot some outfits, but I'm not so fond of the "fight to loot" thing. As has been pointed out repeatedly, if the boarding ship knows where the outfit is, it can blast a whole straight through the ship to access it.

In terms of the looting battle mechanic, what it sounds to me is we need some kind of "Strategic Control" boarding option. As in an attack option that instead of trying to capture the whole ship, it is just trying to gain control of certain key sections such as Engineering. It'd be pretty much a normal attack action, except only targeting about a quarter of the target crew. Once complete it would give access to the Reactor and other "internal" outfits.

My proposal would be to have a series of steps/options for boarding:

 - Initially, one would have the option of looting all the "external" outfits, and presented with buttons to Attack, Defend, Temporarily raid engineering, and extract a target outfit.

   - Attack and Defend works the same as they do now, and have the goal of taking the ship.

   - Temporarily raid engineering engages in a series of attack actions that only face a quarter of the crew,  and results in clearing a path to engineering. This would "unlock" looting all the outfits marked "internal." (The downside of doing this instead of attacking would be that doing this the player risks loosing crew in combat as per normal, but the target ship suffers 50% fewer casualties)

   - Extract target outfit (requires an Outfit Scanner) would give you the option to loot any one outfit from the list marked "internal" without needing to fight the crew, but the ship is destroyed in the process, so no further looting or combat is possible.

**Comment by endless-sky** [comment]
As with every design choice in the game, the plundering / capturing mechanics are not based on what is the most realistic, but on what is the most fun - which also includes game balance and comprehensibility. The only reason to make the mechanics more complicated would be to either improve game balance or to introduce a change that makes new and interesting strategic choices available to the player.

**Comment by Zitchas** [comment]
Very true. I'm not sure if any of these suggestions (mine included) would make it more fun, although I'd like to think that it would make it more interesting. I would certainly enjoy it, but then I like complexity, too. I don't know how universal that is.

One thing I've noticed is that the Outfit Scanner is pretty much useless as far as I can tell, with the exception of a handful of missions that require it. There is no combat application/benefit, and little enough variety in ship outfitting that it isn't useful in picking targets. So having a tiered boarding system as described would enable it to have an additional use for those inclined to board and loot for targetted high value outfits.

A side consequence of this system would be to make raiding Raiders and World Ships a higher risk activity, since the player wouldn't be able to loot as much of the ship without actually fighting the crew. If they opt for the "Extract target outfit" they'd be able to get one outfit of interest, but wouldn't be able to get it all. So when faced with choices such as, say, looting a Jump Drive or a Large System Core, they'd have to pick one or the other instead of taking both. Or decide to take their chances with at least a limited boarding action. 

This might be a case where additional options make things both more realistic and more fun. (at least in my view)

**Comment by ravenshining** [comment]
Given how ridiculously profitable plundering currently is, I think this would certainly address balance, while at the same time adding some fun drama to the act of otherwise just picking the most valuable things off a list that is already sorted by value.  Also, addressing the balance in this way might open up the possibility of relaxing FW and merchant aggression against disabled ships, which some find to be strange.

Alternative ways to deal with this imbalance and add drama, for those who prefer to imagine you blasting a hole in the ship to remove components rather than boarding it to remove them, would be to:
- have a % chance for the ship to explode if an item is plundered à la EV, or
- deal a certain amount of damage to the ship if an item is plundered, although this mightn't make sense in light of how you can repair friendly ships.

**Comment by ThrawnCA** [comment]
If #7333 were implemented, then perhaps plundering outfits could cause extra damage to them, compared to capturing the ship intact?

**Comment by Amazinite** [comment]
I don't see this as being something we implement. I can't find the exact quote at the moment, but I believe @Quantumshark has said something to the effect of how plundering is limited by your cargo space and capturing is limited by your bunk space. Having plundering require crew combat would just end up tying both mechanics to your bunk space, and would likely just result in plundering being less engaged with and increase how often people capture. Since capturing the ship gets you all of its outfits, why only plunder when you're already fighting the crew and could go just a little further to capture the whole ship?

**Comment by ThrawnCA** [comment]
> Since capturing the ship gets you all of its outfits, why only plunder when you're already fighting the crew and could go just a little further to capture the whole ship?

The intention is that you could still plunder _some_ outfits. External guns and such would be easy game.

It's only when you want the heart of the ship that it would - and arguably should - become about as hard as fully taking over.
